### PR TITLE
Inspections on redundant field modifiers in @Value class

### PR DESCRIPTION
**Pull request by wyhasany**
It would be nice to inform users that private and final modifiers
on fields are redundant in classes marked with @Value.

Lombok Inspections(1)

// Tests has been added

### CODE REVIEW

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    @Override
+    public void visitField(PsiField field) {
+      super.visitField(field);
+
+      final PsiElement parent = field.getParent();
+      if (parent instanceof PsiClass) {
+        final PsiClass clazz = (PsiClass) parent;
+        if (clazz.hasAnnotation("lombok.Value")) {
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: Please use lombol.Value.class.getName() syntax

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          final PsiModifierList modifierList = field.getModifierList();
+          if (modifierList != null) {
+            if (modifierList.hasExplicitModifier(PRIVATE)) {
+              final Optional<PsiElement> psiPrivate = Arrays.stream(modifierList.getChildren())
+                .filter(psiElement -> psiElement.toString().contains("private"))
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: Better compare PsiModifier constant with text of element:
PRIVATE.equals(psiElement.getText()) should be better

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+              psiPrivate.ifPresent(psiElement -> holder.registerProblem(psiElement,
+                "Redundant private field modifier",
+                ProblemHighlightType.WARNING,
+                new RemoveModifierFix(PRIVATE))
+              );
+            }
+            if (modifierList.hasExplicitModifier(FINAL)) {
+              final Optional<PsiElement> psiFinal = Arrays.stream(modifierList.getChildren())
+                .filter(psiElement -> psiElement.toString().contains("final"))
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: same here

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                .findFirst();
+
+              psiFinal.ifPresent(psiElement -> holder.registerProblem(psiElement,
+                "Redundant final field modifier",
+                ProblemHighlightType.WARNING,
+                new RemoveModifierFix(PRIVATE))
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: copy paste error -> FINAL

Comment (mplushnikov) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: Try to add QuickFix test:-)

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static com.intellij.psi.PsiModifier.FINAL;
+import static com.intellij.psi.PsiModifier.PRIVATE;
+
+/**
+ * @author Plushnikov Michail
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: This is not true any more :-)

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java (hunk 4 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: shame on me :D

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java (hunk 5 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: that's true :+1:

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java (hunk 1 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: fixed

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java (hunk 4 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: that's so interesting :) Everything works flawlessly in Idea even with wrong QuickFix

File: src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java (hunk 2 above)
Comment (wyhasany) [reply src/main/java/de/plushnikov/intellij/plugin/inspection/RedundantModifiersOnValueLombokAnnotationInspection.java]: good approach easy to refactor in future

**Pull request merged**